cargo-fix/tests/all/main.rs

```diff
@@ -0,0 +1,264 @@
+extern crate difference;
+extern crate url;
+
+use std::env;
+use std::fs;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::str;
+use std::sync::atomic::*;
+use std::time::Instant;
+
+use difference::{Changeset, Difference};
+use url::Url;
+
+static CNT: AtomicUsize = ATOMIC_USIZE_INIT;
+thread_local!(static IDX: usize = CNT.fetch_add(1, Ordering::SeqCst));
+
+struct ProjectBuilder {
+    files: Vec<(String, String)>,
+}
+
+struct Project {
+    root: PathBuf,
+}
+
+fn project() -> ProjectBuilder {
+    ProjectBuilder {
+        files: Vec::new(),
+    }
+}
+
+fn root() -> PathBuf {
+    let idx = IDX.with(|x| *x);
+
+    let mut me = env::current_exe().unwrap();
+    me.pop(); // chop off exe name
+    me.pop(); // chop off `deps`
+    me.pop(); // chop off `debug` / `release`
+    me.push("generated-tests");
+    me.push(&format!("test{}", idx));
+    return me
+}
```

Comment:
This whole fn looks like a `tempdir::TempDir::new()` to me. The thread_local index that on init increments an atomic is pretty cool though (took me a while to see what's going on). Still haven't seen why the project dirs ever get cleaned up.

Reply:
Indeed! I learned this when working on other CLI tools that the achilles heel of TempDir is the reason it exists, automatic cleanup. When debugging why a test fails it's often difficult and annoying to set up the workspace again and it's often much easier to just go poking around directly in the `target` directory to see what went wrong